File: Components/WinForms/VB/XpoTutorials/Images/Cars.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports DevExpress.Xpo
Imports DevExpress.Xpo.Metadata

Namespace DevExpress.Xpo.Demos
    Public Class Cars
        Inherits XPLiteObject
        Private fID As Integer
        <Key()> _
        Public Property ID() As Integer
            Get
                Return fID
            End Get
            Set(ByVal value As Integer)
                SetPropertyValue(Of Integer)("ID", fID, value)
            End Set
        End Property
        Private fTrademark As String
        <Size(SizeAttribute.Unlimited)> _
        Public Property Trademark() As String
            Get
                Return fTrademark
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("Trademark", fTrademark, value)
            End Set
        End Property
        Private fModel As String
        <Size(SizeAttribute.Unlimited)> _
        Public Property Model() As String
            Get
                Return fModel
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("Model", fModel, value)
            End Set
        End Property
        Private fHP As Short
        Public Property HP() As Short
            Get
                Return fHP
            End Get
            Set(ByVal value As Short)
                SetPropertyValue(Of Short)("HP", fHP, value)
            End Set
        End Property
        Private fLiter As Double
        Public Property Liter() As Double
            Get
                Return fLiter
            End Get
            Set(ByVal value As Double)
                SetPropertyValue(Of Double)("Liter", fLiter, value)
            End Set
        End Property
        Private fCyl As Byte
        Public Property Cyl() As Byte
            Get
                Return fCyl
            End Get
            Set(ByVal value As Byte)
                SetPropertyValue(Of Byte)("Cyl", fCyl, value)
            End Set
        End Property
        Private fCategory As String
        <Size(SizeAttribute.Unlimited)> _
        Public Property Category() As String
            Get
                Return fCategory
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("Category", fCategory, value)
            End Set
        End Property
        Private fDescription As String
        <Size(SizeAttribute.Unlimited)> _
        Public Property Description() As String
            Get
                Return fDescription
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("Description", fDescription, value)
            End Set
        End Property
        Private fHyperlink As String
        <Size(SizeAttribute.Unlimited)> _
        Public Property Hyperlink() As String
            Get
                Return fHyperlink
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("Hyperlink", fHyperlink, value)
            End Set
        End Property
        Private fIcon() As Byte
        <Size(SizeAttribute.Unlimited)> _
        Public Property Icon() As Byte()
            Get
                Return fIcon
            End Get
            Set(ByVal value As Byte())
                SetPropertyValue(Of Byte())("Icon", fIcon, value)
            End Set
        End Property
        Private fPicture As System.Drawing.Image
        <Size(SizeAttribute.Unlimited), ValueConverter(GetType(DevExpress.Xpo.Metadata.ImageValueConverter))> _
        Public Property Picture() As System.Drawing.Image
            Get
                Return fPicture
            End Get
            Set(ByVal value As System.Drawing.Image)
                SetPropertyValue(Of System.Drawing.Image)("Picture", fPicture, value)
            End Set
        End Property
        Private fPrice As Decimal
        Public Property Price() As Decimal
            Get
                Return fPrice
            End Get
            Set(ByVal value As Decimal)
                SetPropertyValue(Of Decimal)("Price", fPrice, value)
            End Set
        End Property
        Private fDeliveryDate As DateTime
        <Persistent("Delivery Date")> _
        Public Property DeliveryDate() As DateTime
            Get
                Return fDeliveryDate
            End Get
            Set(ByVal value As DateTime)
                SetPropertyValue(Of DateTime)("DeliveryDate", fDeliveryDate, value)
            End Set
        End Property
        Private fIsInStock As Boolean
        <Persistent("Is In Stock")> _
        Public Property IsInStock() As Boolean
            Get
                Return fIsInStock
            End Get
            Set(ByVal value As Boolean)
                SetPropertyValue(Of Boolean)("IsInStock", fIsInStock, value)
            End Set
        End Property
        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
        Public Overrides Sub AfterConstruction()
            MyBase.AfterConstruction()
        End Sub
    End Class
End Namespace

File: Components/WinForms/CS/XpoTutorials/Images/Cars.cs
using System;
using System.Drawing;
using DevExpress.Xpo;
using DevExpress.Xpo.Metadata;

namespace DevExpress.Xpo.Demos
{
    public class Cars : XPLiteObject {
        int fID;
        [Key]
        public int ID {
            get { return fID; }
            set { SetPropertyValue<int>("ID", ref fID, value); }
        }
        string fTrademark;
        [Size(SizeAttribute.Unlimited)]
        public string Trademark {
            get { return fTrademark; }
            set { SetPropertyValue<string>("Trademark", ref fTrademark, value); }
        }
        string fModel;
        [Size(SizeAttribute.Unlimited)]
        public string Model {
            get { return fModel; }
            set { SetPropertyValue<string>("Model", ref fModel, value); }
        }
        short fHP;
        public short HP {
            get { return fHP; }
            set { SetPropertyValue<short>("HP", ref fHP, value); }
        }
        double fLiter;
        public double Liter {
            get { return fLiter; }
            set { SetPropertyValue<double>("Liter", ref fLiter, value); }
        }
        byte fCyl;
        public byte Cyl {
            get { return fCyl; }
            set { SetPropertyValue<byte>("Cyl", ref fCyl, value); }
        }
        string fCategory;
        [Size(SizeAttribute.Unlimited)]
        public string Category {
            get { return fCategory; }
            set { SetPropertyValue<string>("Category", ref fCategory, value); }
        }
        string fDescription;
        [Size(SizeAttribute.Unlimited)]
        public string Description {
            get { return fDescription; }
            set { SetPropertyValue<string>("Description", ref fDescription, value); }
        }
        string fHyperlink;
        [Size(SizeAttribute.Unlimited)]
        public string Hyperlink {
            get { return fHyperlink; }
            set { SetPropertyValue<string>("Hyperlink", ref fHyperlink, value); }
        }
        byte[] fIcon;
        [Size(SizeAttribute.Unlimited)]
        public byte[] Icon {
            get { return fIcon; }
            set { SetPropertyValue<byte[]>("Icon", ref fIcon, value); }
        }
        System.Drawing.Image fPicture;
        [Size(SizeAttribute.Unlimited)]
        [ValueConverter(typeof(DevExpress.Xpo.Metadata.ImageValueConverter))]
        public System.Drawing.Image Picture {
            get { return fPicture; }
            set { SetPropertyValue<System.Drawing.Image>("Picture", ref fPicture, value); }
        }
        decimal fPrice;
        public decimal Price {
            get { return fPrice; }
            set { SetPropertyValue<decimal>("Price", ref fPrice, value); }
        }
        DateTime fDeliveryDate;
        [Persistent(@"Delivery Date")]
        public DateTime DeliveryDate {
            get { return fDeliveryDate; }
            set { SetPropertyValue<DateTime>("DeliveryDate", ref fDeliveryDate, value); }
        }
        bool fIsInStock;
        [Persistent(@"Is In Stock")]
        public bool IsInStock {
            get { return fIsInStock; }
            set { SetPropertyValue<bool>("IsInStock", ref fIsInStock, value); }
        }
        public Cars(Session session) : base(session) { }
        public override void AfterConstruction() { base.AfterConstruction(); }
    }
}
